debezium-core/src/test/java/io/debezium/pipeline/source/snapshot/incremental/SignalBasedSnapshotChangeEventSourceTest.java

```diff
@@ -185,6 +293,33 @@ public void testBuildQueryThreePkColumnsWithAdditionalCondition() {
                 "SELECT * FROM \"s1\".\"table1\" WHERE ((\"pk1\" > ?) OR (\"pk1\" = ? AND \"pk2\" > ?) OR (\"pk1\" = ? AND \"pk2\" = ? AND \"pk3\" > ?)) AND NOT ((\"pk1\" > ?) OR (\"pk1\" = ? AND \"pk2\" > ?) OR (\"pk1\" = ? AND \"pk2\" = ? AND \"pk3\" > ?)) AND \"val1\"=foo ORDER BY \"pk1\", \"pk2\", \"pk3\" LIMIT 1024");
     }
 
+    @Test
+    public void testBuildUnionQueryThreePkColumnsWithAdditionalCondition() {
+        final SignalBasedIncrementalSnapshotChangeEventSource<? extends Partition, TableId> source = new SignalBasedIncrementalSnapshotChangeEventSource<>(
+                configWithUnionBasedQuery(), new JdbcConnection(config().getJdbcConfig(), config -> null, "\"", "\""), null, null, null, SnapshotProgressListener.NO_OP(),
+                DataChangeEventListener.NO_OP(), notificationService);
+        final IncrementalSnapshotContext<TableId> context = new SignalBasedIncrementalSnapshotContext<>();
+        source.setContext(context);
+        final Column pk1 = Column.editor().name("pk1").create();
+        final Column pk2 = Column.editor().name("pk2").create();
+        final Column pk3 = Column.editor().name("pk3").create();
+        final Column val1 = Column.editor().name("val1").create();
+        final Column val2 = Column.editor().name("val2").create();
+        final Table table = Table.editor().tableId(new TableId(null, "s1", "table1"))
+                .addColumn(pk1)
+                .addColumn(pk2)
+                .addColumn(pk3)
+                .addColumn(val1)
+                .addColumn(val2)
+                .setPrimaryKeyNames("pk1", "pk2", "pk3").create();
+        assertThat(source.buildChunkQuery(table, Optional.of("\"val1\"=foo")))
+                .isEqualTo("SELECT * FROM \"s1\".\"table1\" WHERE \"val1\"=foo ORDER BY \"pk1\", \"pk2\", \"pk3\" LIMIT 1024");
+        context.nextChunkPosition(new Object[]{ 1, 5 });
+        context.maximumKey(new Object[]{ 10, 50 });
+        assertThat(source.buildChunkQuery(table, Optional.of("\"val1\"=foo"))).isEqualTo(
```

Comment:
Can we rewrite it as something like this (not tested):

```
assertThat(source.buildChunkQuery(table, Optional.of("\"val1\"=foo"))).isEqualTo(
        "SELECT * FROM (" +
            "(SELECT * FROM \"s1\".\"table1\" " +
                "WHERE (\"pk1\" > ?) AND (\"pk1\" <= ?) AND \"val1\"=foo " +
                "ORDER BY \"pk1\", \"pk2\", \"pk3\" LIMIT 1024) " +
            "UNION " +
            "(SELECT * FROM \"s1\".\"table1\" " +
                "WHERE (\"pk1\" = ? AND \"pk2\" > ?) AND (\"pk1\" <> ? OR \"pk2\" <= ?) AND \"val1\"=foo " +
                "ORDER BY \"pk1\", \"pk2\", \"pk3\" LIMIT 1024) " +
            "UNION " +
            "(SELECT * FROM \"s1\".\"table1\" " +
                "WHERE (\"pk1\" = ? AND \"pk2\" = ? AND \"pk3\" > ?) AND (\"pk1\" <> ? OR \"pk2\" <> ? OR \"pk3\" <= ?) AND \"val1\"=foo " +
                "ORDER BY \"pk1\", \"pk2\", \"pk3\" LIMIT 1024)" +
        ") AS t " +
        "ORDER BY \"t\".\"pk1\", \"t\".\"pk2\", \"t\".\"pk3\" LIMIT 1024"
);
```

Reply:
Let me try that, project has it's own formatter, idk how it would handle this case.

Reply:
Fixed this one